### PR TITLE
doc: aws_iot: Update documentation to include topic handling

### PR DESCRIPTION
**Pull request by simensrostad**
Update documentation to include topic handling for subscribing and publishing to custom topics.

Fixes NCSIDB-620

Signed-off-by: Simen S. Røstad <simen.rostad@nordicsemi.no>

### CODE REVIEW

**Comment by simensrostad** [comment]
@divipillai This PR needs a doc review

File: doc/nrf/libraries/networking/aws_iot.rst [hunk 1]
```diff
@@ -154,6 +154,76 @@ To subscribe to non-AWS specific topics, complete the following steps:
...
+	}
+
+Publishing to non-AWS specific topics
+======================================
+
+To publish to a non-AWS specific topic, complete the following steps:
+
+* Populate a :c:struct:`aws_iot_topic_data` with the custom topics that you want to publish to.
+  It is not necessary to set the topic type when populating the :c:struct:`aws_iot_topic_data structure. This type is reserved for AWS IoT shadow topics.
+* Pass in the entry that corresponds to the topic that the payload is to be published to in the message structure :c:struct:`aws_iot_data` that is passed into the :c:func:`aws_iot_send` function.
```
Comment (divipillai) [inline doc/nrf/libraries/networking/aws_iot.rst:190]:
```suggestion
* Pass in the entry that corresponds to the topic that the payload is to be published to in the message structure :c:struct:`aws_iot_data`.
  This structure is then passed into the :c:func:`aws_iot_send` function.
```

File: doc/nrf/libraries/networking/aws_iot.rst [hunk 2]
```diff
@@ -154,6 +154,78 @@ To subscribe to non-AWS specific topics, complete the following steps:
...
+
+To publish to a non-AWS specific topic, complete the following steps:
+
+* Populate a :c:struct:`aws_iot_topic_data` with the custom topics that you want to publish to.
+  It is not necessary to set the topic type when populating the :c:struct:`aws_iot_topic_data` structure.
+  This type is reserved for AWS IoT shadow topics.
+* Pass in the entry that corresponds to the topic that the payload is to be published to in the message structure :c:struct:`aws_iot_data`.
+  This structure is then passed into the :c:func:`aws_iot_send` function.
+
+The following code example shows how to Publishing to non-AWS specific topics:
```
Comment (divipillai) [inline doc/nrf/libraries/networking/aws_iot.rst:194]:
```suggestion
The following code example shows how to publish to non-AWS specific topics:
```

**Review by divipillai — APPROVED** [review]

**Review by didrikr — APPROVED** [review]
Looks good

**Review by rlubos — APPROVED** [review]

**Pull request merged**